### PR TITLE
Added Kryo integration to provide backward-compatibility

### LINKED ISSUE #22: Ensure backwards-compatibility for instances.dat file
We need to provide backwards-compatibility for the file that we keep the instance information.

### PR DESCRIPTION
**Pull request by alparslanavci**
Fixes #22.

### CODE REVIEW

File: src/main/java/com/hazelcast/commandline/member/ProcessStore.java [hunk 1]
```diff
@@ -75,21 +80,20 @@ void save(HazelcastProcess process)
...
             return processes;
         }
-        try (FileInputStream fileInputStream = new FileInputStream(instancesFilePath);
-             ObjectInputStream input = new ObjectInputStream(fileInputStream)) {
-            processes = (Map<String, HazelcastProcess>) input.readObject();
-        } catch (ClassNotFoundException cnfe) {
-            throw new HazelcastException(cnfe.getMessage(), cnfe);
+        try (FileInputStream fileInputStream = new FileInputStream(instancesFilePath)) {
+            Input input = new Input(fileInputStream);
+            processes = kryo.readObject(input, HashMap.class);
```
Comment (mesutcelik) [inline src/main/java/com/hazelcast/commandline/member/ProcessStore.java]: Why it his HashMap?

File: src/main/java/com/hazelcast/commandline/member/ProcessStore.java [hunk 2]
```diff
@@ -75,21 +80,20 @@ void save(HazelcastProcess process)
...
+            input.close();
```
Comment (mesutcelik) [inline src/main/java/com/hazelcast/commandline/member/ProcessStore.java]: Do we need `input.close` as Input is AutoCloseable
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
https://github.com/EsotericSoftware/kryo/blob/master/src/com/esotericsoftware/kryo/io/Input.java#L33

Comment (alparslanavci) [reply src/main/java/com/hazelcast/commandline/member/ProcessStore.java]: Yes, we don't need it. I will also change the output part with try-with-resources.

**Comment by alparslanavci** [comment]
> I added some comments. Can we also add some tests covering the changed parts?

I am awaiting other test structure PR to be merged to add the tests for this code. After the merge, I will add them.

**Pull request merged**